actionsystemtest/lib/action_system_test/test_helpers.rb

```diff
@@ -0,0 +1,9 @@
+module ActionSystemTest
+  module TestHelpers
+    extend ActiveSupport::Autoload
+
+    autoload :ActiveJobSetup
```

Comment:
@jnicklas (I didn't see this answered elsewhere. My apologies if it was)

1. this `autoload` is actually not Module::autoload, but one added by extending `ActiveSupport::Autoload` and is somewhat incompatible. The Rails version uses dependency tracking to ensure what only what is needed is booted, and that what that needs is also booted :)
1. The Ruby Module.autoload has been threadsafe since sometime in 2.0 across all Ruby impls. For convenience, pasting some discussion from https://github.com/headius/thread_safe/issues/15#issuecomment-28928923

> Why does the `thread_safe` gem use [autoload](https://github.com/headius/thread_safe/blob/1773812/lib/thread_safe.rb#L5) which is known not to be thread-safe? (Or so I understand)
> - In 2008 @headius wrote in [http://bugs.ruby-lang.org/issues/show/921](http://bugs.ruby-lang.org/issues/show/921) and [https://www.ruby-forum.com/topic/172385](https://www.ruby-forum.com/topic/172385)

> > Currently autoload is not safe to use in a multi-threaded application. To put it more bluntly, it's broken.
> > 
> > The current logic for autoload is as follows:
> > 1. A special object is inserted into the target constant table, used as a marker for autoloading
> > 2. When that constant is looked up, the marker is found and triggers autoloading
> > 3. The marker is first removed, so the constant now appears to be undefined if retrieved concurrently
> > 4. The associated autoload resource is required, and presumably redefines the constant in question
> > 5. The constant lookup, upon completion of autoload, looks up the constant again and either returns its new value or proceeds with normal constant resolution
> > 
> > The problem arises when two or more threads try to access the constant. Because autoload is stateful and unsynchronized, the second thread may encounter the constant table in any number of states:
> > 1. It may see the autoload has not yet fired, if the first thread has encountered the marker but not yet removed it. It would then proceed along the same autoload path, requiring the same file a second time.
> > 2. It may not find an autoload marker, and assume the constant does not exist.
> > 3. It may see the eventual constant the autoload was intended to define.
> > 
> > Of these combinations, (3) is obviously the desired behavior. (1) can only happen on native-threaded implementations that do not have a global interpreter lock, since it requires concurrency during autoload's internal logic. (2) can happen on any implementation, since while the required file is processing the original autoload constant appears to be undefined.
> - And in 2011, @matz said [https://www.ruby-forum.com/topic/3036681](https://www.ruby-forum.com/topic/3036681) 
  
  > > 'autoload will be dead, I strongly discourage the use of autoload in any standard libraries'

The @thedarkone wrote that `autoload` has been thread-safe since about 2012. So, I looked it up:

> @thedarkone  So, the internet is wrong inasmuch as  `autoload` is thread-safe in 
> - cruby >= [2.0](http://bugs.ruby-lang.org/projects/ruby-trunk/repository/revisions/33078), [diff](https://github.com/ruby/ruby/commit/cfe8ed5cbc763a3abcfc19ce10619836aeff7682)
>   - https://bugs.ruby-lang.org/projects/ruby-trunk/repository/revisions/33078
>   -[aws confirms autoload is now safe](https://ruby.awsblog.com/post/Tx16QY1CI5GVBFT/Threading-with-the-AWS-SDK-for-Ruby) is Ruby 2.0 by @lsiegel
> - rbx some patch level
> - jruby some patch level 
>   - [jruby issue where autoload made thread-safe](http://jira.codehaus.org/browse/JRUBY-3194)
> - Discussion [1](https://github.com/rkh/rack-protection/issues/45), [2](https://www.ruby-forum.com/topic/4414805), [3](http://blog.plataformatec.com.br/2012/08/eager-loading-for-greater-good/)